packages/editor/src/components/local-autosave-monitor/index.js

```diff
@@ -88,45 +62,74 @@ function useAutosaveNotice() {
 	const {
 		postId,
 		getEditedPostAttribute,
-	} = useSelect( ( select ) => ( {
-		postId: select( 'core/editor' ).getCurrentPostId(),
-		getEditedPostAttribute: select( 'core/editor' ).getEditedPostAttribute,
-	} ) );
+		remoteAutosave,
+		hasFetchedAutosave,
+	} = useSelect( ( select ) => {
+		const _postId = select( 'core/editor' ).getCurrentPostId();
+		const postType = select( 'core/editor' ).getCurrentPostType();
+		const user = select( 'core' ).getCurrentUser();
+		return {
+			postId: _postId,
+			getEditedPostAttribute: select( 'core/editor' ).getEditedPostAttribute,
+			remoteAutosave: select( 'core' ).getAutosave( postType, _postId, user.id ),
+			hasFetchedAutosave: select( 'core' ).hasFetchedAutosaves( postType, _postId ),
+		};
+	} );
 
 	const { createWarningNotice, removeNotice } = useDispatch( 'core/notices' );
 	const { editPost, resetEditorBlocks } = useDispatch( 'core/editor' );
 
 	useEffect( () => {
-		let autosave = window.sessionStorage.getItem( postKey( postId ) );
-		if ( ! autosave ) {
+		if ( ! hasFetchedAutosave ) {
+			return;
+		}
+
+		let localAutosave = localAutosaveGet( postId );
+		if ( ! localAutosave ) {
 			return;
 		}
 
 		try {
-			autosave = JSON.parse( autosave );
+			localAutosave = JSON.parse( localAutosave );
 		} catch ( error ) {
 			// Not usable if it can't be parsed.
 			return;
 		}
 
-		const { post_title: title, content, excerpt } = autosave;
+		const { post_title: title, content, excerpt } = localAutosave;
 		const edits = { title, content, excerpt };
 
-		// Only display a notice if there is a difference between what has been
-		// saved and that which is stored in sessionStorage.
-		const hasDifference = Object.keys( edits ).some( ( key ) => {
-			return edits[ key ] !== getEditedPostAttribute( key );
-		} );
-
-		if ( ! hasDifference ) {
-			// If there is no difference, it can be safely ejected from storage.
-			window.sessionStorage.removeItem( postKey( postId ) );
+		{
```

Comment:
Is this needed? a way to create a scope?

Reply:
Yes, it was for scope, and to share the WIP quickly